pkg/apis/core/validation/shoot.go

```diff
@@ -571,7 +571,27 @@ func validateKubernetes(kubernetes core.Kubernetes, fldPath *field.Path) field.E
 			}
 		}
 
-		allErrs = append(allErrs, ValidateWatchCacheSizes(kubeAPIServer.WatchCacheSizes, fldPath.Child("watchCacheSizes"))...)
+		allErrs = append(allErrs, ValidateWatchCacheSizes(kubeAPIServer.WatchCacheSizes, fldPath.Child("kubeAPIServer", "watchCacheSizes"))...)
+
+		const maxMaxRequestsInflight = 800
```

Comment:
Don't set a limit - the API server can handle way above that number with better etcd tuning.

Reply:
What about the concerns from @timuthy in https://github.com/gardener/gardener/issues/3108#issuecomment-721782228?

> the API server can handle way above that number with better etcd tuning.

Yes, but do we have this etcd tuning already? If we don't set a limit then we can expect people to set this to a very high number, and this will definitely result in scalability issues and ops effort. I would really like to limit this, at least at the beginning, to be on the safe side. It's already possible to set it twice as high as the default which should be a good first improvement. I'm open to further increase the limit in case we see that we can cater with it.

OK?

Reply:
Here is what K8S is doing:

https://github.com/kubernetes/kubernetes/blob/c7911a384cbc11a4b5003da081b181d6b814d07e/test/kubemark/resources/start-kubemark-master.sh#L566-L570

For 1000 Nodes max requests in flight is 1500 and mutating 500
For 3000 Nodes max requests in flight is 3000 and mutating 3000

Reply:
Thanks for the reference. I still suggest to start with above values, we can increase them later if needed.